docs/manual/format.md

```diff
@@ -264,3 +256,101 @@ could start at byte 589, byte that is an improper boundary for an INT32.
 As a result, 3 null bytes are inserted and the date for the SIZE actually
 starts at byte 592: "00 09 9b 31", which is 629553).
 
+### Immutable header regions
```

Comment:
I kept any changes to this document in a separate commit so they can be more easily reviewed.

There's plenty of room for improvement still, but this seems halfway reasonable?

Reply:
There's lots in the hregion text that fits neither the style or content otherwise (like lack of technical detail), but it's a start. Documenting the details is going to be quite a bit of work that I need to do anyhow sooner than later.